diff --git a/BUILD.bazel b/BUILD.bazel
index 7c479ca..3df076f 100644
--- a/BUILD.bazel
+++ b/BUILD.bazel
@@ -36,12 +36,15 @@ cc_library(
         "include/oneapi/tbb/detail/*.h",
     ]),
     copts = ["-w"] + select({
+        "@platforms//os:emscripten": ["-fexceptions", "-pthread"],
         "@platforms//os:windows": [""],
-        "@platforms//cpu:arm64": [""],
-        "//conditions:default": ["-mwaitpkg"],
+        "@platforms//os:linux": ["-mwaitpkg"], # FIXME: would fail on non x86_64
+        "@platforms//os:macos": [""],
+        "//conditions:default": [""],
     }),
     defines =
         select({
+            "@platforms//os:emscripten": ["__EMSCRIPTEN__", "TBB_EMSCRIPTEN=1", "__TBB_RESUMABLE_TASKS=0", "USE_PTHREAD", "TBB_USE_EXCEPTIONS=0", "TBB_STRICT=0", "TBB_DISABLE_HWLOC_AUTOMATIC_SEARCH=1", "BUILD_SHARED_LIBS=0", "__TBB_EMSCRIPTEN_STACK_SIZE=65536", "TBB_EXAMPLES=1", "TBB_TEST=1"], # TODO: `local_defines` this
             "@platforms//cpu:x86_64": ["__TBB_NO_IMPLICIT_LINKAGE"],
             "//conditions:default": [
                 "USE_PTHREAD",
@@ -56,6 +59,7 @@ cc_library(
     ],
     linkopts =
         select({
+            "@platforms//os:emscripten": ["-fexceptions", "-pthread", "-ldl", "-sALLOW_MEMORY_GROWTH=1", "-sMALLOC=mimalloc", "-sEXIT_RUNTIME=1", "-sERROR_ON_UNDEFINED_SYMBOLS=0"], # FIXME: `-Wl,--no-undefined` here sounds dangerous
             "@platforms//os:windows": [],
             "@platforms//os:linux": [
                 "-ldl",
